src/library_async.js

```diff
@@ -501,7 +501,14 @@ mergeInto(LibraryManager.library, {
         abort('invalid state: ' + Asyncify.state);
       }
       return Asyncify.handleSleepReturnValue;
-    }
+    },
+
+    handleAsync: function(startAsync) {
```

Comment:
a short comment here about how this differs from `handleSleep` would help I think

Reply:
Yeah, good point. I guess I was reluctant because `handleSleep` doesn't have comment either :P

How about this?

```suggestion
    // Unlike `handleSleep`, accepts a function returning a `Promise`
    // and uses the fulfilled value instead of passing in a separate callback.
    //
    // This is particularly useful for native JS `async` functions where the
    // returned value will "just work" and be passed back to C++.
    handleAsync: function(startAsync) {
```

Reply:
Looks good!